### ISSUE TITLE
mesa-18-musl-pthread.patch: logical errors

### ISSUE THREAD

**Issue by saih792** [gentoo/musl#173]
```
defined(__linux__) && !(defined(__GLIBC__) || (__GLIBC__ < 3 || (__GLIBC__ == 2 && __GLIBC_MINOR__ < 12)))
```
Should instead be:
```
defined(__linux__) && !(defined(__GLIBC__) && (__GLIBC__ < 3 && (__GLIBC__ != 2 || __GLIBC_MINOR__ < 12)))
```

It seems that someone forgot to exchange operators when applying De Morgan's law to the original expression; in my opinion it would be better to use the expanded form to avoid similar mistakes:
```
defined(__linux__) && (!defined(__GLIBC__) || __GLIBC__ >= 3 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 12))
```

**Comment by blueness** [comment]
I have to admit, I copied the patch without much thought.  I'll get the fix in with later.

**Comment by blueness** [comment]
fixed in c574022b7d9ecd047be2c82223736b953b8f7d23
